### ISSUE TITLE
[1.1.1-1.20.1]  Outdated structures cause DFU to process every time they are loaded

### ISSUE THREAD

**Issue by Linguardium** [ZsoltMolnarrr/Gazebo#1]
You have outdated structure files in your mod. This causes minecraft to try to update them every time they are loaded, leading to extra memory and cpu usage during worldgen. Please update your structures to match the mc version you are releasing for.

This should be as easy as loading them into game and re-saving them or using a mod to load them into a StructureTemplate and writing them back out to nbt.

**Comment by ZsoltMolnarrr** [comment]
Hi!

Thank you for the feedback.
I think I missed this technical change, could you point me to any technical page (like Fabric wiki / changelog / whatever) that  explains what you say?
Reexporting structures takes a lot more time than you might think, since the Air/Void block placement is lost during the process, which is tedious to build.

**Comment by Linguardium** [comment]
yeah, thats why i also mentioned StructureTemplate read/writing which is how DFU does it.

https://minecraft.wiki/w/Data_version
here is a what i am currently using to test and output datapacks, though i removed some secondary purpose code (like checking loot tables in structures) and its not complete since it comes from a larger method that iterates all datapacks in the environment https://gist.github.com/Linguardium/247448473633829838032a0f44ff6993

You can check out StructureTemplate to see what it does to update as well
